### PR TITLE
[TIMOB-19666] iOS 9.1: Support PHLivePhoto + PHLivePhotoView

### PR DESCRIPTION
**Pull request by hansemannn**
JIRA: https://jira.appcelerator.org/browse/TIMOB-19666

Demo:

``` javascropt
var window = Ti.UI.createWindow({
  title: "iOS 9.1 Live Photos",
  backgroundColor: "#fff"
});

var view = null;

var nav = Ti.UI.iOS.createNavigationWindow({
  window: window
});

var btn1 = Ti.UI.createButton({
  systemButton: Ti.UI.iPhone.SystemButton.ADD,
  top:30
});

btn1.addEventListener("click", openGallery);

var btn2 = Ti.UI.createButton({
  systemButton: Ti.UI.iPhone.SystemButton.TRASH,
  top:10
});

btn2.addEventListener("click", resetImageView);

var btn3 = Ti.UI.createButton({
  title: "Start livePhoto",
  visible: false,
  bottom:10
});

btn3.addEventListener("click", startLivePhoto);

window.setRightNavButton(btn1);
window.setLeftNavButton(btn2);
window.add(btn3);

nav.open();

function openGallery() {
  Ti.Media.openPhotoGallery({
    mediaTypes: [Ti.Media.MEDIA_TYPE_PHOTO, Ti.Media.MEDIA_TYPE_LIVEPHOTO],
    success: function(e) {
      var photo = e.media; // Static photo without motion of type Ti.Blob
      var livePhoto = e.livePhoto // Live photo of type Ti.UI.iOS.LivePhoto

      if(livePhoto) {
        // Live photo supported and returned
        view = Ti.UI.iOS.createLivePhotoView({
          livePhoto: livePhoto,
          muted: true,
          width: 300
        });

        view.addEventListener("start", function(e) {
            Ti.API.warn("Start playback");
            Ti.API.warn(e);
        });

        view.addEventListener("stop", function(e) {
            Ti.API.warn("Stop playback");
            Ti.API.warn(e);
        });

      } else if(photo) {
        // Default photo for iOS < 9.1
        view = Ti.UI.createImageView({
          image: photo
        });
      } else {
        Ti.API.error("Fallback: Most probably tried to select a video (not part of this example).");
      }

      btn3.setVisible(livePhoto != null);

      window.add(view);
    }
  });
}

function resetImageView() {
    if(!view) {
        return;
    }

    window.remove(view);
}

function startLivePhoto() {
    if(!view) {
        return;
    }

    view.startPlaybackWithStyle(Ti.UI.iOS.LIVEPHOTO_PLAYBACK_STYLE_FULL);

    view.setMuted(false);
    setTimeout(function() {
        view.stopPlayback();
    }, 1000);
}
```

### CODE REVIEW

File: iphone/Classes/TiUIiOSLivePhoto.h [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/**
+ * Appcelerator Titanium Mobile
+ * Copyright (c) 2009-2015 by Appcelerator, Inc. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License
+ * Please see the LICENSE included with this distribution for details.
+ */
+#import "TiProxy.h"
+#import <Photos/Photos.h>
```
Comment (cheekiatng) [inline iphone/Classes/TiUIiOSLivePhoto.h]: #if IS_XCODE_7_1

File: iphone/Classes/TiUIiOSLivePhoto.m [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ * Copyright (c) 2009-2015 by Appcelerator, Inc. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License
+ * Please see the LICENSE included with this distribution for details.
+ */
+
+#import "TiUIiOSLivePhoto.h"
+
+@implementation TiUIiOSLivePhoto
+
+-(instancetype)initWithLivePhoto:(PHLivePhoto*)livePhoto
```
Comment (cheekiatng) [inline iphone/Classes/TiUIiOSLivePhoto.m]: #if IS_XCODE_7_1

File: iphone/Classes/MediaModule.m [hunk 3]
```diff
@@ -1715,11 +1738,13 @@ - (void)imagePickerController:(UIImagePickerController *)picker_ didFinishPickin
         mediaType = (NSString*)kUTTypeImage; // default to in case older OS
     }
     BOOL isVideo = [mediaType isEqualToString:(NSString*)kUTTypeMovie];
+    BOOL isLivePhoto = ([TiUtils isIOS9_1OrGreater] == YES && [mediaType isEqualToString:(NSString*)kUTTypeLivePhoto]);
```
Comment (cheekiatng) [inline iphone/Classes/MediaModule.m]: #if IS_XCODE_7_1

File: iphone/Classes/MediaModule.m [hunk 4]
```diff
@@ -1715,11 +1738,13 @@ - (void)imagePickerController:(UIImagePickerController *)picker_ didFinishPickin
...
     
     NSURL *mediaURL = [editingInfo objectForKey:UIImagePickerControllerMediaURL];
 	
     NSDictionary *cropRect = nil;
     TiBlob *media = nil;
+    TiUIiOSLivePhoto *livePhoto = nil;
```
Comment (cheekiatng) [inline iphone/Classes/MediaModule.m]: #if IS_XCODE_7_1

File: iphone/Classes/MediaModule.m [hunk 5]
```diff
@@ -1782,6 +1807,10 @@ - (void)imagePickerController:(UIImagePickerController *)picker_ didFinishPickin
             UISaveVideoAtPathToSavedPhotosAlbum(tempFilePath, nil, nil, NULL);
         }
     }
+    else if(isLivePhoto) {
```
Comment (cheekiatng) [inline iphone/Classes/MediaModule.m]: #if IS_XCODE_7_1

File: iphone/Classes/MediaModule.m [hunk 6]
```diff
@@ -1847,9 +1876,13 @@ - (void)imagePickerController:(UIImagePickerController *)picker_ didFinishPickin
     [dictionary setObject:mediaType forKey:@"mediaType"];
     [dictionary setObject:media forKey:@"media"];
 
-    if (thumbnail!=nil) {
+    if (thumbnail != nil) {
         [dictionary setObject:thumbnail forKey:@"thumbnail"];
     }
+    
+    if (livePhoto != nil) {
```
Comment (cheekiatng) [inline iphone/Classes/MediaModule.m]: #if IS_XCODE_7_1

File: iphone/Classes/TiUIiOSLivePhotoView.h [hunk 7]
```diff
@@ -0,0 +1,20 @@
+/**
+ * Appcelerator Titanium Mobile
+ * Copyright (c) 2009-2015 by Appcelerator, Inc. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License
+ * Please see the LICENSE included with this distribution for details.
+ */
+#import "TiUIView.h"
+#import "TiUIiOSLivePhoto.h"
+#import <PhotosUI/PhotosUI.h>
```
Comment (cheekiatng) [inline iphone/Classes/TiUIiOSLivePhotoView.h]: #if IS_XCODE_7_1

File: iphone/Classes/TiUIiOSLivePhotoView.m [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+ * Appcelerator Titanium Mobile
+ * Copyright (c) 2009-2015 by Appcelerator, Inc. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License
+ * Please see the LICENSE included with this distribution for details.
+ */
+
+#import "TiUIiOSLivePhotoView.h"
+#import "TiUIiOSLivePhotoViewProxy.h"
+
+@implementation TiUIiOSLivePhotoView
```
Comment (cheekiatng) [inline iphone/Classes/TiUIiOSLivePhotoView.m]: #if IS_XCODE_7_1

File: iphone/Classes/TiUIiOSLivePhotoViewProxy.h [hunk 9]
```diff
@@ -0,0 +1,34 @@
+/**
+ * Appcelerator Titanium Mobile
+ * Copyright (c) 2009-2015 by Appcelerator, Inc. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License
+ * Please see the LICENSE included with this distribution for details.
+ */
+#import "TiViewProxy.h"
+#import "TiUIiOSLivePhoto.h"
+#import "TiUIiOSLivePhotoView.h"
+#import <PhotosUI/PhotosUI.h>
```
Comment (cheekiatng) [inline iphone/Classes/TiUIiOSLivePhotoViewProxy.h]: #if IS_XCODE_7_1

File: iphone/Classes/TiUIiOSLivePhotoViewProxy.m [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+ * Appcelerator Titanium Mobile
+ * Copyright (c) 2009-2015 by Appcelerator, Inc. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License
+ * Please see the LICENSE included with this distribution for details.
+ */
+
+#import "TiUIiOSLivePhotoViewProxy.h"
+#import "TiUtils.h"
+
+@implementation TiUIiOSLivePhotoViewProxy
```
Comment (cheekiatng) [inline iphone/Classes/TiUIiOSLivePhotoViewProxy.m]: #if IS_XCODE_7_1

File: apidoc/Titanium/UI/iOS/LivePhotoView.yml [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+extends: Titanium.UI.View
+platforms: [iphone,ipad]
+since: "5.2.0"
+osver: {ios:{min: "9.1"}}
+
+properties:
+  - name: muted
+    summary: Determines if the playback should be muted (`true`) or not (`false`).
+    type: Boolean
+
```
Comment (cheekiatng) [inline apidoc/Titanium/UI/iOS/LivePhotoView.yml]: default?

**Comment by cheekiatng** [comment]
need to document these 2 constants:
LIVEPHOTO_PLAYBACK_STYLE_FULL
LIVEPHOTO_PLAYBACK_STYLE_HINT
In App/iOS/iOS.yml

**Comment by cheekiatng** [comment]
Test and works well .Please address comments

**Comment by hansemannn** [comment]
Addressed comments. Also just saw this badging API to indicate live photos. Should we add this in this PR or in another feature ticket?

**Comment by cheekiatng** [comment]
Sure add it

**Comment by hansemannn** [comment]
Added badge options and compile macros. For some reason, the build currently exists without throwing an error, will need to investigate first before we can merge. Docs for the badge options coming after that is resolved.

File: iphone/Classes/TiUtils.h [hunk 12]
```diff
@@ -576,8 +576,13 @@ typedef enum
 +(BOOL)isIOS9OrGreater;
 
 /**
+<<<<<<< HEAD
```
Comment (cheekiatng) [inline iphone/Classes/TiUtils.h]: resolve the conflict here

**Comment by cheekiatng** [comment]
Please address comment. Everything else looks good and ready to go.

**Comment by hansemannn** [comment]
Resolved it. However, the travis build failed because of an temporary Android bug. I would like to test the scons'd build again first before we merge.

**Comment by hansemannn** [comment]
@cheekiatng Scons'd & tested, ready to go!!

**Comment by cheekiatng** [comment]
CR and FT passed. APPROVED!

**Pull request merged**